### PR TITLE
closes #65 local dataframe storing

### LINKED ISSUE #65: Local Dataframe Storing Mechanism
- [x] Files are to be stored locally instead of being exchanged with the client on each request.
- [x] Approach should support up to 200k entries.
- [x] API should be generalized to make cloud storage extension implementation easier.

### PR DESCRIPTION
**Pull request by Tremirre**
I implemented the local storing mechanism. It works by assigning a UUID to each user when they upload assays and on every request retrieving data based on that UUID that is stored in the user's browser LocalStorage. When user reenters the dashboard after upload, the data will be loaded back again - provided he did not clear the LocalStorage.

This approach is basically limitless when it comes to number of compounds that it can handle (it is the computation of projections that suffers greatly for larger datasets).

To test whether it works fine for 200k compounds I wrote a simple script that generates minimal assays (in terms of number of columns) of given size and count - I saved it in the repo in case someone also wishes to test some functionality with larger number of compunds under `scripts/generate_mock_assay.py`. In general, any helper scripts we may use during the development should be placed under `scripts` directory.

With the script I generated 5 assays with 200k compounds (each assay representing different experiment). Below is a screenshot with mock data loaded (for testing I disabled UMAP and TSNE calculations as it would take too much time to calculate):

200k_compunds

I wrote a small test suite for LocalFileStorage class, but I did not test any callbacks or preprocessing code, since in the following weeks they are going to change substantially and as such these tests would be virtually useless.

### CODE REVIEW

**Comment by Tremirre** [comment]
+ note that I added new dependency (`pyarrow` for handling parquet files) so you will need to install it after this feature will be merged

File: dashboard/storage/__init__.py [hunk 1]
```diff
@@ -0,0 +1,2 @@
+from .base import FileStorage
+from .local import LocalFileStorage
```
Comment (zuzg) [inline dashboard/storage/__init__.py:2]: no newline

File: dashboard/storage/base.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+class FileStorage(abc.ABC):
+    @abc.abstractmethod
+    def read_file(self, name: str) -> bytes:
+        ...
+    
+    @abc.abstractmethod
+    def save_file(self, name: str, content: bytes) -> None:
+        ...
+
+
```
Comment (zuzg) [inline dashboard/storage/base.py:12]: no newline + empty lines (?)

**Review by zuzg — COMMENTED** [review]
Nice! I added 2 smol comments

File: dashboard/storage/__init__.py (hunk 1 above)
Comment (Tremirre) [reply dashboard/storage/__init__.py]: fixed

File: dashboard/storage/base.py (hunk 2 above)
Comment (Tremirre) [reply dashboard/storage/base.py]: fixed v2

**Review by zuzg — APPROVED** [review]

**Review by azywot — APPROVED** [review]
LGTM (ran locally with mocked compounds) 👍

**Review by AndrzejKaj — APPROVED** [review]

**Pull request merged**